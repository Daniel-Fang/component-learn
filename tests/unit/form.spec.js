import { shallowMount } from '@vue/test-utils';
import FyForm from '@/components/form';

describe('test form', () => {
  it('test form', () => {
    const defaultSlot = 'default slot';
    const wrapper = shallowMount(FyForm, {
      propsData: {
        model: {
          name: ''
        },
        rules: {
          name: [{
            required: true,
            message: '姓名是必填项',
            trigger: 'blur'
          }]
        }
      },
      slots: {
        default: defaultSlot
      }
    });
    
    expect(wrapper.text()).toBe(defaultSlot);
    expect(wrapper.html()).toMatchSnapshot();
  })
})