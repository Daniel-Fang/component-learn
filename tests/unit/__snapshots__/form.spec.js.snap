// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test form test form 1`] = `<div class="fy-form">default slot</div>`;
